src/metabase/query_processor/middleware/permissions.clj

```diff
@@ -83,7 +83,8 @@
       (throw (perms-exception required-perms))))
   ;; check perms for any Cards referenced by this query (if it is a native query)
   (doseq [{query :dataset_query} (qp.resolve-referenced/tags-referenced-cards outer-query)]
-    (check-query-permissions* query)))
+    ;; TODO: review needed:
```

Comment:
@dpsutton This needs attention.

Reply:
@camsaul this seems correct to me but it makes me think this code is not hit at all. I think you were saying the same in slack.

Reply:
Yeah, we discussed this on Slack. I stared at this for a while and concluded that the code here is unused. It's actually a redundant check anyway since if you have ad-hoc native query permissions you also have permissions to run any query against that Database which would mean you would be able to run the queries associated with these Cards anyway. That would probably explain why nothing broke without this change.

I think fixing it here makes sense and we can open an issue to either remove this code entirely or do something else with it